openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -452,7 +481,7 @@ def set_course_cohort_settings(course_key, **kwargs):
         A CourseCohortSettings object.
 
     Raises:
-        ValueError if course_key is invalid.
+        Http404 if course_key is invalid.
```

Comment:
Actually in master all functions are using `get_course_by_id()` which raises Http404. In the first PR of the feature branch this function used `get_course()` and so raised a `ValueError`. This change is to revert to the original behaviour.